pkg/agent/agentserver/server.go

```diff
@@ -84,6 +89,48 @@ var _ agent.AgentServiceServer = &ProxyServer{}
 
 var _ agent.ProxyServiceServer = &ProxyServer{}
 
+func (s *ProxyServer) addFrontend(agentID string, connID int64, p *ProxyClientConnection) {
+	s.fmu.Lock()
```

Comment:
Similar to add/removeBackend, can we add a log line when we add and remove frontends?